coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/attachment/ProposalDevelopmentAttachmentHelper.java

```diff
@@ -38,6 +44,163 @@ public void reset() {
+	       		documentNarratives.remove(documentNarratives.indexOf(updatedNarrative));
+	       		documentNarratives.add(narrative);
+	       	}
+	   	}
+    }
+    	
+    private void removeDeletedNarrativesFromDocument( List<Narrative> formNarratives, List<Narrative> documentNarratives) {
+    	// Delete Narratives which are not on the form - from document.
+    	if(documentNarratives != null && documentNarratives.size() > 0) {
+    		List<Narrative> deletedNarratives = new ArrayList<Narrative>();
+    		for(Narrative narrative : documentNarratives ) {
+    			if(findNarrativesByType(formNarratives, narrative.getNarrativeTypeCode()).size() == 0 ) {
+            		deletedNarratives.add(narrative);
+            	}
+    		}
+	   		if(deletedNarratives.size() > 0) {
+	   			documentNarratives.removeAll(deletedNarratives);
+	   		}
+    	}
+    }
+    
+    protected ProposalPersonBiography findUpdatedPersonBio(List<ProposalPersonBiography> origBios, final ProposalPersonBiography newBio) {
+    	
+    	ProposalPersonBiography updatedBio = CollectionUtils.find(origBios, new Predicate<ProposalPersonBiography>() {
+
+			@Override
+			public boolean evaluate(ProposalPersonBiography input) {
+				if(input.getObjectId().equals(newBio.getObjectId())) {
+					return true;
+				}	
+				else {
+					return false;
+				}
+			}
+		});
+
+    	return updatedBio;
+    }
+    
+    protected Narrative findUpdatedNarrative(List<Narrative> origNarratives, final Narrative newNarrative) {
+    	
+    	Narrative updatedNarrative = CollectionUtils.find(origNarratives, new Predicate<Narrative>() {
+
+			@Override
+			public boolean evaluate(Narrative input) {
+				if(input.getNarrativeTypeCode().equals(newNarrative.getNarrativeTypeCode())) {
+					return true;
+				}	
+				else {
+					return false;
+				}
+			}
+		});
+
+    	return updatedNarrative;
+    }
+    
+    protected List<Narrative> findNarrativesByType(List<Narrative> origNarratives, final String narrativeTypeCode) {
+    	final List<Narrative> narrativesByType = new ArrayList<Narrative>();
+    	CollectionUtils.forAllDo(origNarratives, new Closure<Narrative>() {
+
+			@Override
+			public void execute(Narrative input) {
+
+				if(input.getNarrativeTypeCode().equals(narrativeTypeCode)) {
+					narrativesByType.add(input);	
+				}				
+				
+			}
+
+    	});
+    	return narrativesByType;
+    }
+    
+    public List<ProposalPersonBiography> findPersonBiosByObjectId(List<ProposalPersonBiography> origBios,final ProposalPersonBiography personBiography) {
+    	final List<ProposalPersonBiography> biosByPersonAndType = new ArrayList<ProposalPersonBiography>();
+    	CollectionUtils.forAllDo(origBios, new Closure<ProposalPersonBiography>() {
+
+			@Override
+			public void execute(ProposalPersonBiography input) {
+
+				if(input.getObjectId().equals(personBiography.getObjectId())) {
+					biosByPersonAndType.add(input);	
+				}				
```

Comment:
Closure here too, when you use find up above. Lets use just one. And here you use objectId, which in general we don't make common use of in KC. Better I think to use the primary key which would be proposalPersonNumber and biographyNumber in this case.